entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local filetype="$1"
+  shift
+  local filenames=("$@")
+  matching_files=()
+
+  for file in "${filenames[@]}"; do
+    filename=$(basename "$file")
+    
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function lint_files() {
+  if [[ "${linter[filetype]}" = "all" ]]; then
+    cmd="${linter[name]} ${linter[args]}"
+    eval "$cmd" >> "${linter[logfile]}" 2>&1
+    return
+  fi
+  
+  for file in $(get_files_matching_filetype "${linter[filetype]}" "${included[@]}"); do 
+    if [[ "${linter[executor]+x}" ]]; then
+      cmd="${linter[executor]} ${linter[name]} ${linter[args]} ${file}"
+    else
+      cmd="${linter[name]} ${linter[args]} ${file}"
+    fi
+    eval "$cmd" >> "${linter[logfile]}" 2>&1
+  done
+}
+
+linter_failed="false"
+linter_failures=()
+linter_successes=()
+
 function seiso_lint() {
+  echo -e "\nRunning Seiso Linter\n--------------------------\n"
+
   excluded=()
   included=()
 
   while read -r file; do
-    # Apply filter with =~ to ensure it is aligned with github/super-linter
     if [[ -n ${INPUT_EXCLUDE:+x} && "${file}" =~ ${INPUT_EXCLUDE} ]]; then
       excluded+=("${file}")
       continue
     fi
-
     included+=("${file}")
-
-    # Check Dockerfiles
-    if [[ "${file}" = *Dockerfile ]]; then
-      dockerfile_lint -f "${file}" -r /etc/opt/goat/oci.yml
+  done < <(find . -path "./.git" -prune -o -type f)
+  
+  declare -gA pids
+
+  input="/etc/opt/goat/linters.json"
+
+  while read -r line; do  
+    unset linter
+    declare -gA linter
+ 
+    while IFS='=' read -r key value; do
+      value=$(echo "$value" | tr -d "'")
+      linter["$key"]=$value
+    done < <(echo "$line" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]')
```

Comment:
You are now a `jq` ninja.

![](https://media4.giphy.com/media/3ohhwytHcusSCXXOUg/giphy.gif?cid=ecf05e47mbjjuciyphxs8lqe0moiivcce2f310cmcy915vxa&ep=v1_gifs_search&rid=giphy.gif&ct=g)